core/integration/module_java_test.go

```diff
@@ -288,6 +288,84 @@ func (JavaSuite) TestConstructor(_ context.Context, t *testctx.T) {
 	})
 }
 
+func (JavaSuite) TestEnum(_ context.Context, t *testctx.T) {
+	t.Run("can use an enum value", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		out, err := javaModule(t, c, "enums").
+			With(daggerCall("print", "--severity=LOW")).
+			Stdout(ctx)
+
+		require.NoError(t, err)
+		require.Equal(t, "LOW", out)
+	})
+
+	t.Run("can not use a value not defined in the enum", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		_, err := javaModule(t, c, "enums").
+			With(daggerCall("print", "--severity=FOO")).
+			Stdout(ctx)
+
+		require.Error(t, err)
```

Comment:
Ideally, the error should include the list of possible values for the enums https://docs.dagger.io/api/enumerations/
Maybe you can test that behaviour here for consistency with other SDKs?

Reply:
Fixed, added the line after this one 👍